Color coded steps

magopian opened the pull request:
[user story](https://www.notion.so/Regroupement-des-tapes-transformation-pour-le-textile-b0d409f149b041bbaba94828621db792)

The idea is to use color codes to visualize the different steps accross food and textile.

magopian commented:
Caution: this PR is set on top of #355

@n1k0 not sure yet if that solution answers to the problem described in the user story, i'll wait for feedback from @PascalDagras .

Also, not sure it's worth it to create a new "component" for the border style, or if we should instead use css classes. The pro of using this solution is that we can reuse the color codes defined in `Data.Color`.

n1k0 commented:
@magopian being color blind myself, I'm sure you'll understand why I'm slightly against this visual identification strategy :)

magopian commented:
> @magopian being color blind myself, I'm sure you'll understand why I'm slightly against this visual identification strategy :)

100% agree with that, we don't want the only information to be color! The information is already there (see the naming that's been changed in #355 to spell `transformation` in front of the relevant step names in the textile simulator).

This is reusing the color codes we already have in the comparator to add some coherence between the comparator, the steps tab in the right, and the steps headers on the left.
I'm also thinking about adding a "scroll to" when clicking on the right on the step name, to scroll the left part.

n1k0 commented on src/Data/Textile/Step/Label.elm:
```diff
@@ -38,6 +41,63 @@ all =
+            "materials-step"
+
+        Spinning ->
+            -- We only want a single "transform-step" id
```
Nit: what would be useful is to explain *why* we only want a single one

n1k0 reviewed (approved):
👍

magopian commented:
@n1k0 : thanks for the feedback, I believe i've addressed it. I've also reverted the API "step labels" to return the previous strings (without the `transformation -` prefix).

n1k0 commented on src/Data/Impact.elm:
```diff
@@ -153,7 +153,18 @@ noStepsImpacts =
 
 
-stepsColors : { materials : String, transform : String, packaging : String, transports : String, distribution : String, usage : String, endOfLife : String }
+type alias StepsColors =
+    { materials : String
+    , transform : String
+    , packaging : String
+    , transports : String
+    , distribution : String
+    , usage : String
+    , endOfLife : String
+    }
```
Now if you look at the `StepImpacts`, it feels like we could abstract the type to accept a type variable.

n1k0 replied:
Something like this:

```diff
diff --git a/src/Data/Impact.elm b/src/Data/Impact.elm
index 5dd8d9fa..2a3b5742 100644
--- a/src/Data/Impact.elm
+++ b/src/Data/Impact.elm
@@ -130,17 +130,21 @@ totalComplementsImpactAsChartEntry complementsImpacts =
 -- Steps impacts
 
 
-type alias StepsImpacts =
-    { materials : Maybe Unit.Impact
-    , transform : Maybe Unit.Impact
-    , packaging : Maybe Unit.Impact
-    , transports : Maybe Unit.Impact
-    , distribution : Maybe Unit.Impact
-    , usage : Maybe Unit.Impact
-    , endOfLife : Maybe Unit.Impact
+type alias Steps a =
+    { materials : a
+    , transform : a
+    , packaging : a
+    , transports : a
+    , distribution : a
+    , usage : a
+    , endOfLife : a
     }
 
 
+type alias StepsImpacts =
+    Steps (Maybe Unit.Impact)
+
+
 noStepsImpacts : StepsImpacts
 noStepsImpacts =
     { materials = Nothing
@@ -154,14 +158,7 @@ noStepsImpacts =
 
 
 type alias StepsColors =
-    { materials : String
-    , transform : String
-    , packaging : String
-    , transports : String
-    , distribution : String
-    , usage : String
-    , endOfLife : String
-    }
+    Steps String
 
 
 stepsColors : StepsColors
```

magopian replied:
Yeah, makes sense, adding this, then merging. Thanks!

n1k0 reviewed (commented):
I've commented on the commit, my bad ([link](https://github.com/MTES-MCT/ecobalyse/pull/358/commits/4851a3b990ac87e87db2dd54100c8e9359a15f7c#r1354320428))
